packages/dredd-transactions/compile/compileURI/validateParams.js

```diff
@@ -5,32 +5,46 @@ module.exports = function validateParams(params) {
     let text;
     const param = params[paramName];
 
+    
     if (param.required && !(typeof param.example !== 'undefined' && param.example !== '') && !(typeof param.default !== 'undefined' && param.default !== '')) {
       text = `Required URI parameter '${paramName}' has no example or default value.`;
       result.errors.push(text);
     }
 
-    switch (param.type) {
-      case 'number':
-        if (Number.isNaN(parseFloat(param.example))) {
-          text = `URI parameter '${paramName}' is declared as 'number' but it is a string.`;
-          result.errors.push(text);
-        }
-        break;
-      case 'boolean':
-        if ((param.example !== 'true') && (param.example !== 'false')) {
-          text = `URI parameter '${paramName}' is declared as 'boolean' but it is not.`;
+    if (param.schema && param.schema.example  && typeof param.example === 'string'){
```

Comment:
The `param` structure does not contain a `schema` property. See the implementation of [`compileParams`](https://github.com/apiaryio/dredd/blob/805a797f29915c33c4c6136e5ff0da28ed735837/packages/dredd-transactions/compile/compileURI/compileParams.js#L39) which the result is [passed to `validateParams`](https://github.com/apiaryio/dredd/blob/805a797f29915c33c4c6136e5ff0da28ed735837/packages/dredd-transactions/compile/compileURI/index.js#L38).

```js
return {
  [name]: {
    required: getRequired(memberElement),
    default: getDefault(valueElement),
    example: getExample(valueElement),
    values: getValues(valueElement),
  },
};
```